### PR TITLE
Document POM customization DSL, add complete Maven publishing sample, and remove superfluous model blocks

### LINKED ISSUE #5094: Document how to publish a Java library to Maven Central (including javadoc, sources, and required POM info) using the maven-publish plugin
Depends on #3348.

### PR DESCRIPTION
**Pull request by marcphilipp**
- Remove the documentation for POM modification, i.e. `withXml()`, completely from the User Guide because its usage is discouraged. Instead, the new dedicated DSL for POM customization is documented.
- Add complete example as the last section of the Maven Publish Plugin chapter that showcases how to put together the pieces documented in detail in the previous sections:
  - publications including POM customization and custom artifacts
  - repositories including conditional URL based on project version
  - signing the published artifacts
- Remove model blocks and explanations why they are needed from POM/Ivy descriptor generator task samples because they are no longer needed since publishing does not rely on DeferredConfigurable anymore.

Resolves #5094.

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/publishingMaven.adoc [hunk 1]
```diff
@@ -297,17 +289,30 @@ At times it is useful to generate a Maven POM file for a module without actually
...
+</sample>
 ++++
 
-All details of the publishing model are still considered in POM generation, including `components`, custom `artifacts`, and any modifications made via `pom.withXml`.
+All details of the publishing model are still considered in POM generation, including `components`, custom `artifacts`, and any modifications made via `pom`.
 
-[NOTE]
-====
-The “`maven-publish`” plugin leverages some experimental support for late plugin configuration, and any `GenerateMavenPom` tasks will not be constructed until the publishing extension is configured. The simplest way to ensure that the publishing plugin is configured when you attempt to access the `GenerateMavenPom` task is to place the access inside a `model` block, as the example above demonstrates.
+[[publishing_maven:complete_example]]
```
Comment (eriwen) [inline subprojects/docs/src/docs/userguide/publishingMaven.adoc]: I like how you prefix the subsection with the chapter.

File: subprojects/docs/src/docs/userguide/publishingMaven.adoc [hunk 2]
```diff
@@ -145,25 +145,17 @@ Maven restricts 'groupId' and 'artifactId' to a limited character set (`[A-Za-z0
 
 The only Unicode values that are explicitly prohibited are '```\```', '```/```' and any ISO control character. Supplied values are validated early in publication.
 
-[[sec:modifying_the_generated_pom]]
-==== Modifying the generated POM
+[[sec:customizing_the_generated_pom]]
+==== Customizing the generated POM
```
Comment (pledbrook) [inline subprojects/docs/src/docs/userguide/publishingMaven.adoc]: I think this was an unnecessary change to the anchor. Any links to `#modifying_the_generated_pom` will no longer work. It's fine to change the section title if it has an explicit anchor, just not the anchor itself.

Comment (marcphilipp) [reply subprojects/docs/src/docs/userguide/publishingMaven.adoc]: Thanks for the feedback! I'll change it back.
